<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View</class>
 <widget class="QMainWindow" name="View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>View</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>39</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 2px solid black;</string>
    </property>
   </widget>
   <widget class="QLabel" name="messageLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>501</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>321</width>
      <height>531</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="mazes">
     <attribute name="title">
      <string>Mazes</string>
     </attribute>
     <widget class="QPushButton" name="mazeOpenFile">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generateMazeButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate Maze</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="mazeWidth">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>121</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="mazeHeight">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>110</y>
        <width>121</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>width</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>90</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>height</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QWidget" name="widget_2" native="true">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>250</y>
        <width>261</width>
        <height>131</height>
       </rect>
      </property>
      <widget class="QPushButton" name="buildRouteButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>100</y>
         <width>261</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Build Route</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="fromX">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>44</width>
         <height>26</height>
        </rect>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="fromY">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>70</y>
         <width>44</width>
         <height>26</height>
        </rect>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="toX">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>70</y>
         <width>44</width>
         <height>26</height>
        </rect>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="toY">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>70</y>
         <width>44</width>
         <height>26</height>
        </rect>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>from</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>30</y>
         <width>101</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>to</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>50</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>50</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>50</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Caves</string>
     </attribute>
     <widget class="QPushButton" name="caveOpenFile">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="caveWidth">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>81</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="caveHight">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>110</y>
        <width>81</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>90</y>
        <width>61</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>width</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>90</y>
        <width>61</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>height</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="generateCaveButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate Cave</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="probability">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>110</y>
        <width>81</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>90</y>
        <width>81</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>probability</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="getNextGenButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>270</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Next Generation</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="caveLifeLimit">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>121</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>7</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="caveDeathLimit">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>240</y>
        <width>121</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>7</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>220</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>life limit</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>220</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>death limit</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="autoNextGenSpeed">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>350</y>
        <width>261</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>2000</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>1000</number>
      </property>
     </widget>
     <widget class="QPushButton" name="autoGetNextGenButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>380</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Auto Update</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>330</y>
        <width>261</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>time delay (miliseconds)</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="stopAutoGenButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>410</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>850</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
